sonar-batch/src/main/java/org/sonar/batch/report/TestExecutionAndCoveragePublisher.java

```diff
@@ -105,36 +104,36 @@ public CoverageDetail apply(@Nonnull String testName) {
     }
   }
 
-  private final ResourceCache resourceCache;
+  private final BatchComponentCache componentCache;
   private final TestPlanBuilder testPlanBuilder;
 
-  public TestExecutionAndCoveragePublisher(ResourceCache resourceCache, TestPlanBuilder testPlanBuilder) {
-    this.resourceCache = resourceCache;
+  public TestExecutionAndCoveragePublisher(BatchComponentCache resourceCache, TestPlanBuilder testPlanBuilder) {
+    this.componentCache = resourceCache;
     this.testPlanBuilder = testPlanBuilder;
   }
 
   @Override
   public void publish(BatchReportWriter writer) {
-    for (final BatchResource resource : resourceCache.all()) {
-      if (!resource.isFile()) {
+    for (final BatchComponent component : componentCache.all()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png)Reduce the total number of break and continue statements in this loop to use at most one.[![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS135)